sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -607,6 +607,31 @@ public override async Task<ImmutableList<StackSummary>> ListStacksAsync(Cancella
             return stacks.ToImmutableList();
         }
 
+        /// <inheritdoc/>
+        public override async Task<StackDeployment> ExportStackAsync(string stackName, CancellationToken cancellationToken = default)
+        {
+            var commandResult = await this.RunCommandAsync(
+                new [] { "stack", "export", "--stack", stackName, "--show-secrets" },
+                cancellationToken);
+            return StackDeployment.FromJsonString(commandResult.StandardOutput);
+        }
+
+        /// <inheritdoc/>
+        public override async Task ImportStackAsync(string stackName, StackDeployment state, CancellationToken cancellationToken = default)
+        {
+            var tempFileName = Path.GetTempFileName();
+            try
+            {
+                File.WriteAllText(tempFileName, state.Json.GetRawText());
+                await this.RunCommandAsync(new [] { "stack", "import", "--file", tempFileName },
+                                           cancellationToken);
```

Comment:
For consistency with other methods in the class using await consider:
```suggestion
                await this.RunCommandAsync(new [] { "stack", "import", "--file", tempFileName },
                                           cancellationToken).ConfigureAwait(false);
```